recipes/protobuf/all/conandata.yml

```
@@ -1,25 +1,25 @@
 sources:
+  "3.25.1":
+    url: "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v3.25.1.tar.gz"
+    sha256: "1a2affa2fbad568b9895b72e3c7cb1f72a14bf2501fba056c724dc68c249cd0f"
```

Protobuf versioning is complicated:

https://protobuf.dev/news/2022-05-06/

The major version for C++ changed with v22:

https://protobuf.dev/news/2022-08-03/

```suggestion
  "4.25.1":
    url: "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v4.25.1.tar.gz"
    sha256: "e5b6dac700b6b319fd3329a84ed1594477d645591d7a484657c84cc8f28e28d8"
```

---

Thanks! This will require quite a bit more work than I anticipated. My only real goal was to get the latest version of gRPC to build.
I noticed that #16098 already includes some of the required changes you mention. I'll revert the changes here and create a separate PR based on them.

---

Done: #21622